syntax = "proto3";
package kirito;
option cc_generic_services = true;

message Result {
    int32 errorCode = 1;
    bytes errorMessage = 2;
}

message LoginRequest {
    bytes name = 1;
    bytes password = 2;
}

message LoginResponse {
    Result result = 1;
    bool success = 2;
}

message UserInforRequest {
    int32 userid = 1;
}

message UserInfo {
    bytes name = 1;
    bytes password = 2;
    bytes email = 3;
    bytes phone = 4;
}

message UserInfoResponse {
    Result result = 1;
    UserInfo userInfo = 2;
}

service UserServiceRPC {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetUserInfo(UserInforRequest) returns(UserInfoResponse);
}